### PR TITLE
[EigenSolversApplication] fix relative path for eigensolvers tests

### PR DESCRIPTION
**Pull request by armingeiser**
The "A.mm" file was not found because of the relative paths.

KratosMultiphysics.ReadMatrixMarketMatrix does not seem to give a helpful error in the tests...

closes #2819

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]
OK

File: applications/EigenSolversApplication/tests/test_eigen_direct_solver.py [hunk 1]
```diff
@@ -15,15 +16,23 @@ def _execute_eigen_direct_solver_test(self, class_name, solver_type):
...
 
         a = KratosMultiphysics.CompressedMatrix()
-        
-        KratosMultiphysics.ReadMatrixMarketMatrix('../../../kratos/tests/A.mm', a) # symmetric test matrix
-        
+
+        this_file_dir = os.path.dirname(os.path.realpath(__file__))
+        base_dir = os.path.dirname(os.path.dirname(os.path.dirname(this_file_dir)))
+        matrix_file_path = os.path.join(base_dir, "kratos")
+        matrix_file_path = os.path.join(matrix_file_path, "/tests")
```
Comment (philbucher) [inline applications/EigenSolversApplication/tests/test_eigen_direct_solver.py]: `/` works only on linux afaik
is adding this necessary? From what I see it should also work without (you are already using `os.path.join`)

Comment (armingeiser) [reply applications/EigenSolversApplication/tests/test_eigen_direct_solver.py]: that was a copy/paste error. Thanks for spotting

Comment (armingeiser) [reply applications/EigenSolversApplication/tests/test_eigen_direct_solver.py]: fixed it

**Review by philbucher — DISMISSED** [review]
thx for fixing

File: applications/EigenSolversApplication/tests/test_eigen_direct_solver.py [hunk 2]
```diff
@@ -15,15 +16,23 @@ def _execute_eigen_direct_solver_test(self, class_name, solver_type):
...
         solver = ConstructSolver(settings)
 
         a = KratosMultiphysics.CompressedMatrix()
-        
-        KratosMultiphysics.ReadMatrixMarketMatrix('../../../kratos/tests/A.mm', a) # symmetric test matrix
-        
+
+        this_file_dir = os.path.dirname(os.path.realpath(__file__))
+        base_dir = os.path.dirname(os.path.dirname(os.path.dirname(this_file_dir)))
+        matrix_file_path = os.path.join(base_dir, "kratos")
```
Comment (philbucher) [inline applications/EigenSolversApplication/tests/test_eigen_direct_solver.py]: the following would also work:
`matrix_file_path = base_dir + os.sep + "kratos + os.sep + "tests" + os.sep + "A.mm"`

Comment (philbucher) [reply applications/EigenSolversApplication/tests/test_eigen_direct_solver.py]: or 
`matrix_file_path = os.path.join(base_dir, "kratos", "tests", "A.mm")`

Comment (armingeiser) [reply applications/EigenSolversApplication/tests/test_eigen_direct_solver.py]: i like the one line option with join

**Review by philbucher — APPROVED** [review]
ok if it works :)

**Comment by oberbichler** [comment]
I used the relative paths because I run the tests with `runkratos test_EigenSolvers...py`. What is the recommented way to test just a specific application?

**Comment by armingeiser** [comment]
It is still using relative paths, but now starting from `test_EigenSolvers...py` file, and not anymore from the current working directory.
The tests on the nightly build are run from kratos/python_scripts and then the relative path was not correct anymore.

**Comment by oberbichler** [comment]
👍 I was just confused by the concatenated `dirname`...

File: applications/EigenSolversApplication/tests/test_eigen_direct_solver.py (hunk 1 above)
Comment (oberbichler) [reply applications/EigenSolversApplication/tests/test_eigen_direct_solver.py]: > `/` works only on linux afaik

Works for Windows too

**Pull request merged**